src/Authoring/TestDiagnostics/ArrayParameterTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Runtime.InteropServices;
+
+namespace TestDiagnostics
+{
+    /* TODO: 
+     * what happens if you put another random attribute on an array param? 
+     * check in WRC3 project ...
+    */
+    public class ReadOnlyArray : System.Attribute
+    {
+        public ReadOnlyArray() { }
+    }
+
+    public class WriteOnlyArray : System.Attribute
+    {
+        public WriteOnlyArray() { }
+    }
+
+    /*
+    public interface IHaveAMethodNamedArray_Valid

```

Comment:
Is all of this commented out due to the attribute or some other reason?

Reply:
Yeah my unit testing story is a little funky -- but your comment about making contexts is spot-on, I think that will improve it a lot.